### PR TITLE
Add askHow, tellHow, untellHow and unachieve illocutions.

### PR DESCRIPTION
**Pull request by javipalanca**
Some illocutions to ask for a plan, tell a plan or untell a plan have been added.
Some examples were also written to illustrate this.
PEP8 styling applied to some files.

### CODE REVIEW

**Comment by niklasf** [comment]
Thanks! Looks mostly good, but I don't like how annotations are parsed, but then converted back to strings and parsed again at runtime, with ad-hoc string searches and slicing.

**Comment by javipalanca** [comment]
Thanks @niklasf 
We'll work in fixing this as soon as posible.

**Comment by javipalanca** [comment]
Hi @niklasf 
Please, would you check if this version meets better the quality criteria of the project?
If so, it would be ready for the merge.

**Comment by javipalanca** [comment]
this is just a kind reminder :)

File: agentspeak/runtime.py [hunk 1]
```diff
@@ -408,37 +415,187 @@ def call(self, trigger, goal_type, term, calling_intention, delayed=False):
...
+        """
+            AskHow is a performative that allows the agent to ask for a plan to another agent.
+            We look in the plan.list of the slave agent for the plan that master wants,
+            if we find it: master agent use tellHow to tell the plan to slave agent
+        """
+        # Receive the agent that ask for the plan
+        for annotation in list(term.annots):
+            if isinstance(annotation, str):
+                if "askHow_sender" in annotation:
+                    sender_name = annotation.split("(")[1].split(")")[0]
```
Comment (niklasf) [inline agentspeak/runtime.py]: Looks fragile.

File: agentspeak/runtime.py [hunk 2]
```diff
@@ -408,37 +415,187 @@ def call(self, trigger, goal_type, term, calling_intention, delayed=False):
...
+
+        plans_to_delete = []
+        plans = self.plans.values()
+        for plan in plans:
+            for plan_instance in plan:
+                if len(plan_instance.annotation) > 0:
+                    if ("@" + str(plan_instance.annotation[0].functor)).startswith(label):
+                        plans_to_delete.append(plan_instance)
+        for plan_instance in plans_to_delete:
+            plan.remove(plan_instance)
```
Comment (niklasf) [inline agentspeak/runtime.py]: `plan` as in the last plan in `plans`?

File: agentspeak/runtime.py [hunk 3]
```diff
@@ -546,6 +711,45 @@ def run(self):
...
+    if isinstance(plan.context, type(TrueQuery())):
+        context = "true"
+    else:
+        context = plan.context
+    
+    body = plan.str_body
+    head = str(plan.head)
+    start = 0
+    
+    if "_X_" in head:
```
Comment (niklasf) [inline agentspeak/runtime.py]: `_X_` can occur in the string representation without any special meaning.

File: agentspeak/runtime.py [hunk 4]
```diff
@@ -546,6 +711,45 @@ def run(self):
...
+                head = head[:first_open+1] + str(plan.args[0]) + head[first_close:]
+            start = head.find(")", start) +1 
+            
+        if plan.args[1] != None:
+            first_open, first_close = head.find("(",start), head.find(")", start)
+            if  "(" in str(plan.args[1]):
+                head = head[:first_open+1] + str(plan.args[1]).split("(")[1].split(")")[0] + head[first_close:]
+            else:
+                head = head[:first_open+1] + str(plan.args[1]) + head[first_close:]
+            start = head.find(")", start) +1
```
Comment (niklasf) [inline agentspeak/runtime.py]: Similarly, ad-hoc parsing like this doesn't work. For example `(` could be part of a string literal, without any special meaning.

File: agentspeak/parser.py [hunk 5]
```diff
@@ -900,26 +902,35 @@ def parse_event(tok, tokens, log):
...
         tok, annotation = parse_literal(tok, tokens, log)
         plan.annotations.append(annotation)
 
     tok, event = parse_event(tok, tokens, log)
     plan.event = event
     plan.loc = event.loc
 
+    # If we find a () in the event this indicate that the trigger plan have arguments, we save them in plan.args[0]
+    if "(" in str(event):
+        plan.args[0] = str(event).split("(")[1].split(")")[0]
```
Comment (niklasf) [inline agentspeak/parser.py:917]: Same.

**Review by niklasf — COMMENTED** [review]
Thanks for the reminder (and your patience). I think the new version is better, but there's still quite a bit of ad-hoc parsing that breaks on more complex inputs.

**Comment by javipalanca** [comment]
Hi @niklasf 
Check if these commits match your quality requirements. If you need more changes don't hesitate to ask!
Thanks.

**Comment by javipalanca** [comment]
Dear @niklasf ,
Just another kind reminder :)
Do you think it would be possible to review these changes in a not too distant future? It would be very interesting for us to be able to release a new version of this package to integrate it in our multi-agent system.
Thank you very much!

**Pull request merged**